searchcore/src/vespa/searchcore/proton/server/lid_space_compaction_job_take2.cpp

```diff
@@ -34,19 +35,19 @@ CompactionJob::scanDocuments(const LidUsageStats &stats)
                 moveDocument(meta, std::make_shared<DoneContext>(std::make_pair(std::move(opsTracker), std::move(onDone))));
             }));
             if (failed) return false;
+            _startedCount.fetch_add(1, std::memory_order_relaxed);
             if (isBlocked(BlockedReason::OUTSTANDING_OPS)) {
                 return true;
             }
         }
     }
-    if (!_scanItr->valid()) {
-        sync();
-    }
     return false;
 }
 
 void
 CompactionJob::moveDocument(const search::DocumentMetaData & meta, std::shared_ptr<IDestructorCallback> context) {
+    _executedCount.fetch_add(1, std::memory_order_relaxed);
```

Comment:
`_executedCount` should be increased after queuing a task on the `_master` executor (in the normal case) and in the early returns. Otherwise, `onStop()` might return after a bucket executor has increased `_executedCount` but before it has queued the task, and `MaintenanceController::killJobs()` might queue the task to drop held jobs too early. The task eventually queued by the bucket executor could then access freed memory.

Reply:
Agree, added a guard that should count at the right places.